### PR TITLE
Cleanup import codepath

### PR DESCRIPTION
**Pull request by labkey-jeckels**
Don't creating an inner transaction to help improve error reporting, fixing a problem Vagisha was seeing.

Pretty big refactor of the previously massive (400+ line) importSkylineDoc() method.

Tests are happy!

### CODE REVIEW

File: src/org/labkey/targetedms/SkylineDocImporter.java [hunk 1]
```diff
@@ -233,465 +234,472 @@ private void importSkylineDoc(TargetedMSRun run, File f) throws XMLStreamExcepti
...
+            }
+            if(optimizationSettings._dpPredictor != null && skyReplicate.getDpPredictor() != null && optimizationSettings._dpPredictor.equals(skyReplicate.getDpPredictor()))
+            {
+                replicate.setDpPredictorId(optimizationSettings._dpPredictor.getId());
+            }
 
-                    if (folderType == TargetedMSService.FolderType.QC)
+            if (folderType == TargetedMSService.FolderType.QC)
+            {
+                // In QC folders insert a replicate only if at least one of the associated sample files will be inserted
```
Comment (vagisha) [inline src/org/labkey/targetedms/SkylineDocImporter.java]: This comment should be removed, I think.  We now keep replicates and sample files from the latest uploaded Skyline document and delete older ones.  This comment (and the one on line 1939) look like they are from before this change.

File: src/org/labkey/targetedms/SkylineDocImporter.java [hunk 2]
```diff
@@ -233,465 +234,472 @@ private void importSkylineDoc(TargetedMSRun run, File f) throws XMLStreamExcepti
...
 
-                // Store the document settings
-                // 0. iRT information
-                run.setiRTscaleId(insertiRTData(parser));
+            // Store the document settings
+            // 0. iRT information
+            run.setiRTscaleId(insertiRTData(parser));
 
+            // 1. Transition settings
+            OptimizationSettings optimizationSettings = insertTransitionSettings(parser.getTransitionSettings());
```
Comment (vagisha) [inline src/org/labkey/targetedms/SkylineDocImporter.java]: Perhaps rename OptimizationSettings to OptimizationInfo to be consistent with ReplicateInfo, ModificationInfo etc.?

**Comment by brendanx67** [comment]
I hope that the tests are also comprehensive.

File: src/org/labkey/targetedms/SkylineDocImporter.java (hunk 1 above)
Comment (labkey-jeckels) [reply src/org/labkey/targetedms/SkylineDocImporter.java]: I think those comments are still relevant. While we retain older data, we still need to insert newly added replicates and sample files.

**Pull request merged**